src/proxy/transparent/TransparentUpgradeableProxy.huff

```diff
@@ -0,0 +1,184 @@
+///         any other operation.
+///         /!\
+///
+///         To avoid https://medium.com/nomic-labs-blog/malicious-backdoors-in-ethereum-proxies-62629adf3357[proxy selector
+///         clashing], which can potentially be used in an attack, this contract uses a pattern that implies two things 
+///         that go hand in hand:
+///         
+///         1. If any account other than the admin calls the proxy, the call will be forwarded to the implementation, even if
+///         that call matches one of the admin functions exposed by the proxy itself.
+///         2. If the admin calls the proxy, it can access the admin functions, but its calls will never be forwarded to the
+///         implementation. If the admin tries to call a function on the implementation it will continue to the code of the 
+///         proxy inheritor's MAIN macro.
+///         This means that all the inheritor code of the MAIN macro will be accessible only by the admin.
+///         
+///         These properties mean that the admin account can only be used for admin actions like upgrading the proxy or changing
+///         the admin, so it's best if it's a dedicated account that is not used for anything else. This will avoid headaches due
+///         to sudden errors when trying to call a function from the proxy implementation.
+///         
+///         Our recommendation is for the dedicated account to be an instance of the {ProxyAdmin} contract. If set up this way,
+///         you should think of the `ProxyAdmin` instance as the real administrative interface of your proxy.
+
+/* Imports */
+#include "../ERC1967/ERC1967Upgrade.huff"
+
+/* Interface */
+#define function admin() nonpayable returns (address)
+#define function implementation() nonpayable returns (address)
+#define function changeAdmin(address) nonpayable returns ()
+#define function upgradeTo(address) nonpayable returns ()
+#define function upgradeToAndCall(address,bytes) nonpayable returns ()
+
+/// @dev Constructor
+/// @param implementation The default implementation address
+/// @param admin The default admin address
+/// @param data The data to send to the implementation 
+#define macro CONSTRUCTOR() = takes(0) returns (0) {
+    // Set the admin address
+    0x20                        // [size]
+    0x20 codesize sub           // [offset, size]
+    0x00                        // [mem, offset, size]
+    codecopy                    // []
+    0x00 mload                  // [admin]
+    _SET_ADMIN()
+
+    // Upgrade implementation and call it
+    0x20                        // [size]
+    0x40 codesize sub           // [offset, size]
+    0x00                        // [mem, offset, size]
+    codecopy                    // []
+    0x00 mload                  // [arg]
+    _UPGRADE_TO()
+}
+
+/// @dev Returns the current admin.
+///
+///      NOTE: If this macro is never call outside of this file, only the admin can 
```

Comment:
As long as the MAIN inheritor macro respect the standards of this contract, this macro can only be called by the admin

Remove this NOTE for every macro of this contract.

Reply:
Actually we can juste adjust the comment like this:
```suggestion
///      NOTE: Only the admin can call this macro.
```